### ISSUE TITLE
Comment on SC 1.4.11 Graphics Contrast

### ISSUE THREAD

**Issue by kerstinp** [w3c/wcag21#413]
I really hope this will make it into WCAG 2.1. 

I'm worried that SC 1.4.13 and also SC 1.4.10 can have negative side effects on this much needed SC and want to explain that. 

In some or probably many cases the SC 1.4.11 itself will be hard to meet for diagramms and some types of other images. I assume that authors will often or sometimes meet this SC by using alternate versions like data tables for diagramms or running text for images of text. On this I'm not worried because of possible positive side effects for screenreader users and people with problems in understanding visualized content like diagramms. 

The reason for my comment is: 

In HTML exchanging an image or adding for example a data table as (proper) alternate version for a diagramm with unsufficient contrast even in the aftermath is much more easer than in PDF. 

When a PDF file is published and/or has undergone the design process, all editorial revisions, the tagging just in very rare cases it is realistic that authors/clients will exchange diagramms or other images which are in scope of the SC. Especially not, when – as an example – an annual report or any other document with diagramms is already published as printed version. Also providing alternate version / long description of a diagramm in the same PDF file would be a problem for example out of citation issues.

But meeting this SC is not lost, even not for already published PDF files. PDF provides the possibility for alternate versions for images with unsufficient contrast (for example diagramms) in one or more attached files. This alternate version must of course meet all other SC of the chosen conformance level to pass SC 1.4.11 Graphics contrast. 

The only user agent – I'm focusing on desktop – in which three of the four required adaptions of SC 1.4.13 can be done for text which was filtered out of a PDF file is as far as I know VIP-Reader. VIP-Reader is not displaying file attachments give no hint that attached files are present. This should be verified or falsified by the Working Group.

Considering this the question is: How would it be possible to test wether an alternate version conforms with all SC of the chosen conformance level and therefore passes SC Graphics contrast? 

One could just check wether an alternate version for a graphic without sufficient contrast in a file attachment would conform if the user agent on which SC 1.4.13 seems to rely on (VIP-Reader) would display the attachment. I also see problems in the context of alternate versions with 1.4.10 because Zoom Content can be very tricky for PDF content.

I'm fine with SC 1.4.11 itself but much worried on how to pass WCAG 2.1 with alternate versions in attached files in PDF which I feel is an important technique .

**Edit KP 14.1..2017:**

I want to add that I appreciate this SC when a data table can serve as an conforming alternate version. Without this option I don't see how for example diagramms which can't be changed or where it is not feasible (because of copyright issues, PDF content where a version is already printed and/or tagged etc.) can pass.

**Comment by awkawk** [comment]
We've made substantial changes to each of these SC, we think we have addressed your comment, but please do review the latest version:

Current SC text is:
The visual presentation of the following have a contrast ratio of at least 3:1 against adjacent color(s): (Level AA)
User Interface Components
Visual information used to indicate states and boundaries of active user interface components, except where the appearance of the component is determined by the user agent and not modified by the author.
Graphical objects
Parts of graphics required to understand the content, except when a particular presentation of graphics is essential to the information being conveyed.
